_posts/2016-04-11-growing-developers-at-greenfield.md

```diff
@@ -0,0 +1,84 @@
+---
+layout: post
+title: "Growing Developers at Greenfield"
+description: "Advice for junior developers getting started in their careers"
+category:
+tags: [lessons, advice, career]
+author: tom_chen
+---
+{% include JB/setup %}
+
+![tree]({{ BASE_PATH
+}}/assets/images/growing-developers-at-greenfield/tree.jpeg)
+
+Before I was a developer at Greenfield, I spent several years working in
+academia. After deciding to make the leap into full-time software development
+and completing [Launch Academy](https://www.launchacademy.com/) last year, I was
+anxious to join a company where I could work with and learn from experienced
+developers who were shipping high quality code. When I arrived at Greenfield, I
+was happy to find a team that was not only welcoming and supportive but also
+active in the open source community, passionate about best practices, and as
+excited about building awesome software as I am. Having been at Greenfield for a
+few months now, I'd like to share a couple of lessons I’ve learned that might be
+valuable to others just getting started in their software careers.
+
+### Think Process, Not Product
+
+When you’re just starting out, it’s easy to look around at more experienced
+developers and feel overwhelmed by how much there is to learn. One of the best
+pieces of advice that I got early on was to not stress about how I stack up
+against everyone else and to just focus on my own learning one day at a time. To
+help myself stay focused on what’s important, I’ve kept a journal of what I’m
+learning each day, and recently, we’ve started documenting our TIL’s in a
+[GitHub repo](https://github.com/greenfieldhq/til/issues) as a team. It’s
+helpful to have a record of what I’m learning not only as a reference for future
+me to look back on but also as a way to pause and recognize my successes and
+reflect on what I could do better tomorrow. If you spend all your energy focused
+on what you don’t know, it’s easy to [downplay the progress you're
+making](http://larahogan.me/donuts/). But If you concentrate on the *process* of
+learning something new each day, those small gains will add up to big wins over
+time. [All great developers were beginners](https://changelog.com/189/) at some
+point, and they achieved developer greatness one day a time, just like you.
+
+### Start Before You're Ready
+
+There’s a common misconception about learning, which assumes that learning and
+doing happen in two discrete steps, one after the other. Beginners often fall
+victim to this way of thinking because it can feel safer to focus on reading
+books, working through tutorials, and conceptual learning without getting one’s
```

Comment:
knowledge is "i know how to install the floor. you just lay each plank down
using this formula"
experience is "haha lol omg installing the floor like that will never work"

On Fri, Apr 8, 2016 at 1:26 PM, Vikram Ramakrishnan <
notifications@github.com> wrote:

> In _posts/2016-04-11-growing-developers-at-greenfield.md
> <https://github.com/greenfieldhq/greenfieldhq.github.io/pull/57#discussion_r59060352>
> :
>
> > +[GitHub repo](https://github.com/greenfieldhq/til/issues) as a team. It’s
> > +helpful to have a record of what I’m learning not only as a reference for future
> > +me to look back on but also as a way to pause and recognize my successes and
> > +reflect on what I could do better tomorrow. If you spend all your energy focused
> > +on what you don’t know, it’s easy to [downplay the progress you're
> > +making](http://larahogan.me/donuts/). But If you concentrate on the *process* of
> > +learning something new each day, those small gains will add up to big wins over
> > +time. [All great developers were beginners](https://changelog.com/189/) at some
> > +point, and they achieved developer greatness one day a time, just like you.
> > +
> > +### Start Before You're Ready
> > +
> > +There’s a common misconception about learning, which assumes that learning and
> > +doing happen in two discrete steps, one after the other. Beginners often fall
> > +victim to this way of thinking because it can feel safer to focus on reading
> > +books, working through tutorials, and conceptual learning without getting one’s
>
> Wait, this sounds like what I was confused about before: "understand
> certain concepts about building an app." If you understood those certain
> concepts, wouldn't you be able to build it? Isn't one's lack of knowledge
> and understanding what causes surprises by not anticipating things? Also,
> "difference between knowledge and experience" is really broad. What do you
> mean by that?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/greenfieldhq/greenfieldhq.github.io/pull/57/files/36bedee869a124aed940794662d10d7583bda2d3#r59060352>
>